### ISSUE TITLE
Is perfomence monitoring really enabled on server?

### ISSUE THREAD

**Issue by romash1408** [nuxt-community/sentry-module#500]
[Documentation ](https://sentry.nuxtjs.org/sentry/options/#tracing) says that `tracing` option
> Enables Sentry Performance Monitoring on the [server](https://docs.sentry.io/platforms/node/performance/) and [browser](https://docs.sentry.io/platforms/javascript/guides/vue/performance/) side.

but I don't see any transaction from server side in sentry...

In sources I found [code](https://github.com/nuxt-community/sentry-module/blob/9614795af3a5aa9730a2fbf4346f1eb0f7010059/lib/core/hooks.js#L169-L170) which adds errorHandler and requestHandler, but [sentry documentation](https://docs.sentry.io/platforms/node/guides/express/performance/) for nodejs perfomence says we should use tracingHandler after requestHandler

Also [there is ability](https://docs.sentry.io/platforms/javascript/performance/connect-services/#pageload) to connect ssr with browser transaction throgh meta tags

**Comment by rchl** [comment]
> but I don't see any transaction from server side in sentry...

Yeah, I guess tracing is supported and can be used by manually logging transaction using `startTransaction` API  but automatic-instrumentation is not enabled by default.

It would make sense to look into enabling automatic-instrumentation either by default or as an opt-in but it would have to be checked whether it works with Nuxt because Nuxt technically uses `connect` and not `express` so we'd need to check if it still works fine.

> Also [there is ability](https://docs.sentry.io/platforms/javascript/performance/connect-services/#pageload) to connect ssr with browser transaction throgh meta tags

Good find. Sentry has a lot of "hidden" goodies and it would make sense to investigate whether it's possible to simplify use of those from the module. I'm often not even aware of new things that are being added there.

**Comment by rchl** [comment]
If you want to experiment with using `SentryTracing.Integrations.Express` before this module does anything about it then you should be able to set it up using `serverMiddleware` option in Nuxt Config. In that case the middleware should get inserted after the Sentry.requestHandler and also after some internal Nuxt middlewares but still before the user code so it should be fairly usable.

**Comment by romash1408** [comment]
> It would make sense to look into enabling automatic-instrumentation either by default or as an opt-in

I think that is should be enabled by default with tracing, at least I expected this behaviour especially after reading mentioned documentation :)

> Nuxt technically uses connect and not express

`Sentry.Handlers.tracingHandler` as other handlers already used in this module is framework agnostic: it's arguments are of type `IncomingMessage` and `ServerResponse` from nodejs `http` module. I already tried it and it works!
Now I'm looking into `Sentry.Integrations.Http` from `@sentry/node` that besides creating tracing spans for outgoing requests can also add breadcrumbs to errors from ssr even without tracing enabled, so it would be great to add it in this module too.
I will create pull request as soon as I get acceptable result

**Comment by rchl** [comment]
The https://docs.sentry.io/platforms/javascript/performance/connect-services/#pageload instructions are kinda broken since:
a) I don't know what they mean by `serializeBaggage`. Am I supposed to use `JSON.stringify`? There is no `serializeBaggage` export in Sentry SDK anywhere.
b) The `getBaggage()` method of span does not exist in the SDK. Those instructions appear out of date...

**Comment by rchl** [comment]
> Now I'm looking into `Sentry.Integrations.Http` from `@sentry/node` that besides creating tracing spans for outgoing requests can also add breadcrumbs to errors from ssr even without tracing enabled, so it would be great to add it in this module too.

`Http` is enabled by default by Node SDK so unless it needs extra optoins to make it work, it should just work by default.

**Comment by lforst** [comment]
Yo, one of the maintainers of the Sentry SDK here. The docs are indeed super confusing and broken. I can give you a quick rundown of how to do it here until we fix the docs:

You can look at how we do it for Next.js:

https://github.com/getsentry/sentry-javascript/blob/73a1bbb142ebca5d54b38844d57545c8deac8794/packages/nextjs/src/client/performance.ts#L128-L141

You basically need to do two things:
- Add `traceParentData` of the parent span to the transaction which can be serialized with `span.toTraceParent()` or deserialized with `extractTraceparentData()` from `@sentry/utils`.
- Add `dynamicSamplingContext` of the parent transaction to the transaction metadata. You can get the dynamic sampling context of a transaction with `transaction.getDynamicSamplingContext()`. You can serialize it to a format that is conforms to the baggage http header spec with `dynamicSamplingContextToSentryBaggageHeader()` from `@sentry/utils` and you can turn that header back into DSC with `baggageHeaderToDynamicSamplingContext(baggage)`

**Comment by lforst** [comment]
It sounds far more complicated than it actually is but I get you :D Feel free to ping me if you need help with anything.

**Comment by arkhamvm** [comment]
@rchl Simple workaround that allows distributed tracing:
```
// nuxt.config.ts
plugins: [
	{src: '~/plugins/sentry-trace.js',            mode: 'server'},
],
```
```
// sentry-trace.js
export default function(context) {
	if (process.server) {
		const transaction = require('@sentry/node').getCurrentHub().getScope().getTransaction();

		context.app.head.meta.push({hid: 'sentry-trace', name: 'sentry-trace', content: transaction.toTraceparent()});
	}
}
```

**Comment by rchl** [comment]
Implementing it in #529
